torchtitan/logging.py

```diff
@@ -23,3 +26,26 @@ def init_logger():
 
     # suppress verbose torch.profiler logging
     os.environ["KINETO_LOG_LEVEL"] = "5"
+
+
+class LogLevel(Enum):
+    DEBUG = "DEBUG"
+    INFO = "INFO"
+    WARNING = "WARNING"
+    ERROR = "ERROR"
+    CRITICAL = "CRITICAL"
+
+    @classmethod
+    def from_string(cls, value: str):
+        try:
+            return cls[value.upper()]
+        except KeyError:
+            raise argparse.ArgumentTypeError(f"Invalid log level: '{value}'. "
+                                             f"Choose from {', '.join([lvl.name for lvl in cls])}.")
+
+def validate_log_level(value):
+    try:
+        return LogLevel.from_string(value)
+    except ValueError as e:
```

Comment:
Is there still a need to catch a ValueError after checking a KeyError?

Reply:
Actually, no. Fixed!